File: Ruby/lib/Cultist.rb
=begin
/***********************************/
|                                   | 
|   Javier Izquierdo Vera           |
|   Juan Carlos Gallardo Morales    |
|   2ºBII                           |
|                                   |
|   NapakalakiGame - PDOO           |
|                                   |
/***********************************/
=end

module NapakalakiGame

  require_relative 'card'
  
  class Cultist
    include Card
    
    def initialize(name, gainedlv)
      @name = name
      @gainedLevels = gainedlv
    end
    
    
    def getBasicValue
      @gainedLevels
    end
  
    def getSpecialValue
      getBasicValue*CultistPlayer.totalCultistPlayers
    end
    
  end
  
  
end

File: Java/src/NapakalakiGame/Cultist.java
/************************************
|                                   | 
|   Javier Izquierdo Vera           |
|   Juan Carlos Gallardo Morales    |
|   2ºBII                           |
|                                   |
|   NapakalakiGame - PDOO           |
|                                   |
************************************/

package NapakalakiGame;

public class Cultist implements Card {
    private String name;
    private int gainedLevels;
    
    public Cultist (String name, int gainedLevels){
        this.name = name;
        this.gainedLevels = gainedLevels;
    }
    
    //**
    public String getName(){
        return name;
    }
    //**
    
    @Override
    public int getBasicValue(){
        return gainedLevels;
    }
    
    @Override
    public int getSpecialValue(){
        return getBasicValue()*CultistPlayer.getTotalCultistPlayer();
    }
    
}
